### DISCUSSION TITLE
ImPlotDragToolFlags_NoFit for PlotLines and so forth

### DISCUSSION THREAD

**Discussion by dsantxez** [epezent/implot#312] [Ideas]
![imagen](https://user-images.githubusercontent.com/2100136/148196533-9ca4a915-833b-4723-b618-bd0cca542a02.png)

Hi there,

I am using ImPlots to develop a library for plotting different types of data (mainly lines) and we want to draw like different areas within the plot to show different info (the red and grey areas in the example).

The way to achieve this has been using the DragLine and PlotShadded. The PlotShadded are drawn to the limit of the canvas, that way if the user zooms out or drags the plots, the shaded area keeps drawing till the X or Y limit.

The problem comes when fitting the Y or X axis. The PlotShadded is considered for the fitting and therefore, when zooming out in this case, the fitting doesn't do anything, since the PlotShadded is already fit.

I would be a nice feature to set a flag for each plotting element to tell if they need to be considered for the fitting or not.

Another option would be to have some tools to draw different forms and shapes in the canvas that are not part of the Plots themselves.

Thank you for this project once again, I am getting used to it and I find it very well thought.

All the best,

Denis

PS: I'm open to work in this feature, NoFit or non-plot element drawing as a PR. I would like first your opinion to see where my efforts should go to in the event of developing the feature and the corresponding PR.

**Comment by epezent** [comment]
Hey sorry for the late response. This is in the works in a ongoing branch of mine that I hope to merge soon. Thanks for offering though!
